### PR TITLE
=act Locale unaware method in Helpers.

### PR DESCRIPTION
**Pull request by andreaTP**
This is split from #20406

### CODE REVIEW

File: akka-actor/src/main/scala/akka/util/Helpers.scala [hunk 1]
```diff
@@ -14,7 +14,11 @@ import java.util.Locale
 
 object Helpers {
 
-  val isWindows: Boolean = System.getProperty("os.name", "").toLowerCase(Locale.ROOT).indexOf("win") >= 0
+  implicit class LocaleUnawareString(s: String) {
```
Comment (ktoso) [inline akka-actor/src/main/scala/akka/util/Helpers.scala]: this is costly and magic, please don't add conversions to add methods :-)
just `Helpers.toRootLowerCase` would be good

**Comment by ktoso** [comment]
Could you grep the project for the `.toLowerCase(Locale.ROOT)` and replace those calls with this one?
It's easier to understand a codebase if it's consistent in calling things

File: akka-actor/src/main/scala/akka/util/Helpers.scala (hunk 1 above)
Comment (andreaTP) [inline akka-actor/src/main/scala/akka/util/Helpers.scala]: Sounds good

**Comment by andreaTP** [comment]
done, except on the http module as discussed.

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
LGTM

**Comment by andreaTP** [comment]
fixed

**Comment by patriknw** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, thanks

**Pull request merged**